activerecord/test/cases/relations_test.rb

```diff
@@ -678,6 +678,86 @@ def test_loading_with_one_association_with_non_preload
     assert_equal Post.find(1).last_comment, post.last_comment
   end
 
+  def test_load_associations_with_array
+    posts = Post.all
+    even_posts, odd_posts = posts.partition.each_with_index { |p, i| i.even? }
+
+    assert_queries(1) do
+      Post.load_associations(even_posts, :comments).each { |p| p.comments }
+    end
+    assert_not odd_posts.first.comments.loaded?
+
+    assert_queries(0) do
+      Post.load_associations(even_posts, :comments)
+    end
+  end
+
+  def test_load_associations_with_relation
+    assert_queries(2) do
+      Post.load_associations(Post.all, :comments).each { |p| p.comments }
+    end
+  end
+
+  def test_load_associations_with_invalid_array_elements
+    posts = ["helloworld", "foo bar", "my first post"]
+    assert_raises(ArgumentError) do
+      Post.load_associations(posts, :comments)
+    end
+  end
+
+  def test_load_associations_with_empty_array
+    assert_queries(0) do
+      assert_empty Post.load_associations([], :comments)
+    end
+  end
+
+  def test_load_associations_with_empty_args
+    posts = Post.all
+    even_posts = posts.partition.each_with_index { |p, i| i.even? }
+
+    assert_raises(ArgumentError) do
+      Post.load_associations(even_posts)
+    end
+  end
+
+  def test_load_associations_with_first_record_loaded
+    posts = Post.all.to_a
+    posts.first.comments.to_a
+
+    assert_queries(1) do
+      Post.load_associations(posts, :comments)
+      posts.each { |p| assert p.comments.loaded? }
+    end
+  end
+
+  def test_load_associations_with_first_through_association_loaded
+    posts = Post.all.to_a
+    posts.first.author
+
+    assert_queries(2) do
+      Post.load_associations(posts, :author_categorizations)
+      posts.each { |p| assert p.author_categorizations.loaded? }
+    end
+  end
+
+  def test_load_association_with_duplicated_records
+    posts = [Post.first]
+    posts.push(posts.first.dup.tap { |d| d.id = posts.first.id })
+
+    assert_queries(1) do
+      Post.load_associations(posts, :comments)
+      posts.each { |p| assert p.comments.loaded? }
+    end
+  end
+
+  def test_load_associations_with_different_types
+    records = Author.all.to_a + Post.all.to_a
+    assert_queries(3) do
+      Author.load_associations(records, :comments)
+      records.each { |r| assert r.comments.loaded? }
+    end
+  end
```

Comment:
Nice to know that we support polymorphic association loading as well!

I'm curious what happens if I keep the variables separate:

```ruby
authors, posts = Author.all.to_a, Post.all.to_a

ActiveRecord::Base.load_associations(authors + posts, :comments)

authors.map(&:comments) # Will SQL queries fire here? Or are the comments loaded?
```

Reply:
Comments are loaded, no SQL queries will fire.

Reply:
👍, would it be clearer if we kept the variables separate in the tests and do `authors + posts`?